tests/query_errors/state_not_static/error.txt

```diff
@@ -0,0 +1,8 @@
+Error: 
+    3 | define script bad_state
+    4 | state
+    5 |   system::nanotime()
```

Comment:
i find it sad, that we cannot use the timestamp of initialization for the initial state. It rules out so many cool possible time-based algos we could pull with scripts. With this we need workarounds to express this. :(

Reply:
I don't think there is, and I don't think there should be a workaround. the timestamp it initializes with becomes pointless as there is no time guarantee as to when the compilation of the code happens it'd be a arbritary.

Reply:
So, should we maybe defer evaluation of the state expression to the moment when the first event arrives and the state is not yet initialized?

Reply:
Then we should do the same for args and with to keep it consistent; it means our pipelines are no longer deterministic and replay testing isn't possible any more. With clustering now the same operator will initialize differently based on the system it's one.

I still think that's a bad place to be in.

Reply:
> With clustering now the same operator will initialize differently based on the system it's one.

I honestly don't see a problem with this.

Reply:
But it is fine, lets keep it a compile time constant for now and see if the need arises to wire this up differently.

Reply:
Ja it's always easier to make things more relaxed then harder, I just looked into how to do a init  on first event and it's a code wise PITA too, our `run` function is structured to change the event as a return value so we'd either need to rewrite that whole logic or clone the event (probably cloning wouldn't be soooo terrible for a init but not pretty either)